File: Artist/app/src/main/java/com/example/artist/network/TopArtistsService.java
package com.example.artist.network;


import com.example.artist.models.TopArtistsResponse;
import com.example.artist.models.TopArtistsResponseResult;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;


public interface TopArtistsService {
    @GET("?method=artist.search&format=json")
    Single<TopArtistsResponseResult> getTopArtists(@Query("artist") String user, @Query("api_key") String apiKey);
}

File: Artist/app/src/main/java/com/example/artist/network/TopArtistsService.kt
package com.example.artist.network

import com.example.artist.models.TopArtistsResponseResult
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface TopArtistsService {
    @GET("?method=artist.search&format=json")
    fun getTopArtists(@Query("artist") user: String, @Query("api_key") apiKey: String): Single<TopArtistsResponseResult>
}